devito/types/dimension.py

```diff
@@ -1493,6 +1511,44 @@ def _arg_values(self, *args, **kwargs):
         return {}
 
 
+class VirtualDimension(CustomDimension):
+
+    """
+    Dimension symbol representing a mock iteration space, which as such
+    is eventually ditched by the compiler.
+
+    Mock iteration spaces are used for compilation purposes only, typically
+    to bind objects such as Guards and Syncs to a specific point in the
+    program flow.
+
+    Examples
+    --------
+    To generate nested conditionals within the same loop nest, one may use
+    VirtualDimensions to represent the different branches of the conditionals.
+
+        .. code-block:: C
+
+        for (int i = i_m; i <= i_M; i += 1)
+          if (i < 10)
+            if (i < 5)
+              do A(i);
+            if (i >= 5)
+              do B(i);
+
+    The above code can be obtained by using one VirtualDimension for the
+    `i < 5` conditional and another VirtualDimension for the `i >= 5` conditional.
+    """
+
+    is_Virtual = True
+
+    __rkwargs__ = ('parent',)
+
+    def __init_finalize__(self, name, parent=None):
+        super().__init_finalize__(name, parent=parent,
+                                  symbolic_min=sympy.S.Zero,
+                                  symbolic_max=sympy.S.One)
```

Comment:
would use S.Zero to make it empty but doubt it matters

Reply:
in fact, we're both wrong . In the case above, the size of the loop would be 2, not 1, since the extremes are included. And in your case , if put S.Zero, the size would be 1, not 0 as you wrote. But conceptually I wanted 1, even though it doesn't matter in practice... but if we generated the VirtualDim loop, a size of 1 would guarantee correct code. So I'm applying your suggestion